scripts/scil_visualize_fodf.py

```diff
@@ -0,0 +1,325 @@
+"""
+
+import argparse
+
+import nibabel as nib
+import numpy as np
+
+from dipy.data import get_sphere
+from dipy.core.sphere import Sphere
+from dipy.reconst.shm import sh_to_sf
+from fury import window, actor
+
+from scilpy.io.utils import (add_sh_basis_args, add_overwrite_arg,
+                             assert_inputs_exist, assert_outputs_exist)
+
+# Default background offset when none is specified
+DEFAULT_OFFSET = 0.5
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    # Positional arguments
+    p.add_argument('input_fodf', type=str, help='Input SH image file')
+
+    p.add_argument('slice_index', type=int,
+                   help='Index of the slice to visualize along a given axis')
+
+    # Window configuration options
+    p.add_argument('--win_dims', nargs=2, metavar=('WIDTH', 'HEIGHT'),
+                   default=[768, 768], type=int,
+                   help='The dimensions for the vtk window')
+
+    p.add_argument('--interactor', default='trackball', type=str,
+                   choices={'image', 'trackball'},
+                   help='Specify interactor mode for vtk window')
+
+    p.add_argument('--axis_name', default='axial', type=str,
+                   choices={'axial', 'coronal', 'sagittal'},
+                   help='Name of the axis to visualize.')
+
+    p.add_argument('--silent', default=False, action='store_true',
+                   help='Disable interactive visualization')
+
+    p.add_argument('--output', type=str, help='Path to output file')
+
+    add_overwrite_arg(p)
+
+    # Optional FODF personalization arguments
+    p.add_argument('--sh_order', type=int, default=8,
+                   help='The SH order of the input fODF.')
+
+    add_sh_basis_args(p)
+
+    p.add_argument('--full_basis', default=False, action='store_true',
+                   help='Use full SH basis to reconstruct fODF from '
+                   'coefficients')
+
+    sphere_choices = {'symmetric362', 'symmetric642', 'symmetric724',
+                      'repulsion724', 'repulsion100', 'repulsion200'}
+    p.add_argument('--sphere', default='symmetric724', choices=sphere_choices,
+                   help='Name of the sphere used to reconstruct SF')
+
+    p.add_argument('--sph_subdivide', type=int,
+                   help='Number of subdivisions for given sphere')
+
+    p.add_argument('--mask', help='Optional mask file. Only fODF inside the '
+                                  'mask are displayed')
+
+    p.add_argument('--scale', default=0.5, type=float,
+                   help='Scaling factor for FODF.')
+
+    p.add_argument('--radial_scale_off', default=False, action='store_true',
+                   help='Disable radial scale for ODF slicer')
+
+    p.add_argument('--norm_off', default=False, action='store_true',
+                   help='Disable normalization of ODF slicer')
+
```

Comment:
I would add a `--colormap` argument for the fODF.

Reply:
Good idea, I'll add it :+1: